volatility3/framework/plugins/linux/vmayarascan.py

```diff
@@ -0,0 +1,120 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+from typing import Iterable, List, Tuple
+
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins import yarascan
+from volatility3.plugins.linux import pslist
+
+
+class VmaYaraScan(interfaces.plugins.PluginInterface):
+    """Scans all virtual memory areas for tasks using yara."""
+
+    _required_framework_version = (2, 4, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.BooleanRequirement(
+                name="wide",
+                description="Match wide (unicode) strings",
+                default=False,
+                optional=True,
+            ),
+            requirements.StringRequirement(
+                name="yara_rules", description="Yara rules (as a string)", optional=True
+            ),
+            requirements.URIRequirement(
+                name="yara_file", description="Yara rules (as a file)", optional=True
+            ),
+            # This additional requirement is to follow suit with upstream, who feel that compiled rules could potentially be used to execute malicious code
+            # As such, there's a separate option to run compiled files, as happened with yara-3.9 and later
+            requirements.URIRequirement(
+                name="yara_compiled_file",
+                description="Yara compiled rules (as a file)",
+                optional=True,
+            ),
+            requirements.IntRequirement(
+                name="max_size",
+                default=0x40000000,
+                description="Set the maximum size (default is 1GB)",
+                optional=True,
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.VersionRequirement(
+                name="yarascanner", component=yarascan.YaraScanner, version=(2, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pid",
+                element_type=int,
+                description="Process IDs to include (all other processes are excluded)",
+                optional=True,
+            ),
+        ]
+
+    def _generator(self):
+        rules = yarascan.YaraScan.process_yara_options(dict(self.config))
```

Comment:
If this is processing rules from YaraScan, perhaps it makes sense to use the same requirements as YaraScan, so that changes there get reflected here automatically?  That wouldn't work if there's options in the original that you *don't* want in vmayarascan, but otherwise it feels like it might reduce duplication slightly.

Reply:
Yes, I do want to use the same requirements as YaraScan, just add in an option for a pid too. I'll look into this now.